allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -82,15 +117,50 @@ def tokens_to_indices(self, tokens: List[Token], vocabulary: Vocabulary) -> Inde
                     "Using PretrainedTransformerIndexer but field text_id is not set"
                     f" for the following token: {token.text}"
                 )
+        return indices
 
-        # The mask has 1 for real tokens and 0 for padding tokens. Only real tokens are attended to.
-        mask = [1] * len(indices)
+    def _postprocess_output(self, output: IndexedTokenList) -> IndexedTokenList:
+        """
+        Takes an IndexedTokenList about to be returned by `tokens_to_indices()` and adds any
+        necessary postprocessing, e.g. long sequence splitting.
 
-        return {"token_ids": indices, "mask": mask}
+        The input should have a `"token_ids"` key corresponding to the token indices. They should
+        have special tokens already inserted.
+        """
+        if self._max_length is not None:
+            # We prepare long indices by converting them to (assuming max_length == 5)
+            # [CLS] A B C [SEP] [CLS] D E F [SEP] ...
+            # Embedder is responsible for folding this 1-d sequence to 2-d and feed to the
+            # transformer model.
+            # TODO(zhaofengw): we aren't respecting word boundaries when segmenting wordpieces.
+
+            indices = output["token_ids"]
+            # Strips original special tokens
+            indices = indices[self._num_added_start_tokens : -self._num_added_end_tokens]
+            # Folds indices
+            folded_indices = [
+                indices[i : i + self._effective_max_length]
+                for i in range(0, len(indices), self._effective_max_length)
+            ]
+            # Adds special tokens to each segment
+            folded_indices = [
+                self._tokenizer.build_inputs_with_special_tokens(segment)
+                for segment in folded_indices
+            ]
+            # Flattens
+            indices = [i for segment in folded_indices for i in segment]
+
+            output["token_ids"] = indices
+            output["segment_concat_mask"] = [1] * len(indices)
```

Comment:
Shouldn't this just be the `"mask"` and we can remove `"segment_concat_mask"`? You've already modified `output["token_ids"]` to be a concatenation of segments, so it seems like the mask should correspond to that. Further, in the embedder it seems like the usage of "mask" and "segment_concat_mask" are mutually exclusive.

Reply:
No because the `"mask"` key is reserved by whoever wants to call `get_text_field_mask`, and `"segment_concat_mask"` is for ourself to consume internally and to discard after the unfolding. See https://github.com/allenai/allennlp/pull/3666/files#diff-19a237a1412853c5e07d541a4e2ba518R58-R63